integration_tests/integrationtest.go

```diff
@@ -0,0 +1,609 @@
+	//	return len(l.Items)
+	//})
+	//checkNoObjectsInCluster(&api.SPIAccessCheckList{}, func(l *api.SPIAccessCheckList) int {
+	//	return len(l.Items)
+	//})
+	//checkNoObjectsInCluster(&api.SPIAccessTokenDataUpdateList{}, func(l *api.SPIAccessTokenDataUpdateList) int {
+	//	return len(l.Items)
+	//})
+	//checkNoObjectsInCluster(&api.SPIFileContentRequestList{}, func(l *api.SPIFileContentRequestList) int {
+	//	return len(l.Items)
+	//})
+}
+
+func checkNoObjectsInCluster[L client.ObjectList](list L, getLen func(l L) int) {
+	Eventually(func(g Gomega) {
+		g.Expect(ITest.Client.List(ITest.Context, list)).To(Succeed())
+		g.Expect(getLen(list)).To(BeZero())
+	}).Should(Succeed())
+}
+
+func createAll[T client.Object](objs []T) []T {
+	ret := make([]T, 0, len(objs))
+	for _, proto := range objs {
+		o := proto.DeepCopyObject().(T)
+		Expect(ITest.Client.Create(ITest.Context, o)).To(Succeed())
+		ret = append(ret, o)
+	}
+
+	return ret
+}
+
+func deleteAll[T client.Object](objs []T) {
+	for _, o := range objs {
+		Eventually(func(g Gomega) {
+			cpy := o.DeepCopyObject().(client.Object)
+			err := ITest.Client.Get(ITest.Context, client.ObjectKeyFromObject(o), cpy)
+			// tolerate if the object is no longer found because it was for example deleted during the test
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					g.Expect(err).To(Succeed())
+				}
+				return
+			}
+
+			err = ITest.Client.Delete(ITest.Context, cpy)
+			// tolerate if the object is no longer found because it was for example deleted during the test
+			if err != nil && !errors.IsNotFound(err) {
+				g.Expect(err).To(Succeed())
+			}
+		}).Should(Succeed())
+	}
+}
+
+func waitForStatus[T client.Object, L client.ObjectList](list L, itemsFromList func(L) []T, hasEmptyStatus func(T) bool) {
+	Eventually(func(g Gomega) {
+		g.Expect(ITest.Client.List(ITest.Context, list)).To(Succeed())
+		items := itemsFromList(list)
+		if len(items) == 0 {
+			return
+		}
+
+		for _, o := range items {
+			g.Expect(hasEmptyStatus(o)).To(BeFalse(), "%s/%s/%s has empty status", o.GetObjectKind().GroupVersionKind().String(), o.GetNamespace(), o.GetName())
+		}
+	}).Should(Succeed(), "failed to wait for status of objects %T", list)
+}
+
+func addAllExisting[T client.Object, L client.ObjectList](list L, itemsFromList func(L) []T) []T {
+	var existing []T
+	Expect(ITest.Client.List(ITest.Context, list)).To(Succeed())
+
+	for _, o := range itemsFromList(list) {
+		existing = append(existing, o)
+	}
+
+	return existing
+}
+
+func hasAnyChanged[T client.Object](origs []T, news []T) bool {
+	if len(origs) != len(news) {
+		return true
+	}
+
+	origMap := map[client.ObjectKey]T{}
+	for _, o := range origs {
+		origMap[client.ObjectKeyFromObject(o)] = o
+	}
+
+	for _, n := range news {
+		o := origMap[client.ObjectKeyFromObject(n)]
+		if hasChanged(o, n) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func hasChanged[T client.Object](a T, b T) bool {
```

Comment:
I think the resulting code would be more complex with `deep` than without it, because we would need to base the decision on parsing the returned array of differing field names.

Reply:
Yes, but we're ignoring differences in `annotations`, `resourceVersion` and `managedFields`. If your intention was to have a readable diff, than yes, I am all for it (i.e. just replace `reflect.DeepEqual` with `deep.Equal` and somehow propagate that diff up the call-chain). If your intention was to replace the clearing of the ignored fields and `reflect.DeepEqual` with a call to `deep.Equal` and then parsing the output to ignore the differences in annos, etc, then I would be against.